Sources/Algorithms/Unique.swift

```diff
@@ -30,8 +39,22 @@ extension Sequence where Element: Hashable {
 
 extension Sequence {
   /// Returns an array with the unique elements of this sequence (as determined
-  /// by the given projection), in the order of the first occurence of each
+  /// by the given projection), in the order of the first occurrence of each
   /// unique element.
+  ///
+  ///     let animals = ["dog", "pig", "cat", "ox", "dog","cat"]
+  ///     let uniqued = animals(on: { $0.first })
+  ///     print(uniqued)
+  ///     // Prints "["dog", "pig","cat", "ox"]"
+  ///
+  /// - Parameter projection: A projecting closure. `projection` accepts an
+  ///   element of this sequence as its parameter and returns a projected
+  ///   value of the same  type.
```

Comment:
```suggestion
  ///   value of the same type.
```

This description is true, but it’s also just describing the type signature in words. But what are the semantics expected of this closure? That should be documented here.

Reply:
Added constraint of "Hashable" element